diff --git a/d3d8.h b/d3d8.h
index 347a86c..3af708b 100644
--- a/d3d8.h
+++ b/d3d8.h
@@ -44,7 +44,7 @@
 /*
  * Interface IID's
  */
-#if defined( _WIN32 ) && !defined( _NO_COM)
+#if (defined( _WIN32 ) || defined( __EMSCRIPTEN__ )) && !defined( _NO_COM)
 
 /* IID_IDirect3D8 */
 /* {1DD9E8DA-1C77-4d40-B0CF-98FEFDFF9512} */
diff --git a/d3d8types.h b/d3d8types.h
index 0ee0021..300cb2f 100644
--- a/d3d8types.h
+++ b/d3d8types.h
@@ -20,7 +20,7 @@
 #include <float.h>
 
 #pragma warning(disable:4201) // anonymous unions warning
-#if defined (_WIN32) && !defined (_WIN64)
+#if (defined (_WIN32) || defined (__EMSCRIPTEN__)) && !defined (_WIN64)
 #pragma pack(4)
 #endif
 
@@ -1587,7 +1587,7 @@ typedef struct _D3DADAPTER_IDENTIFIER8
     char            Driver[MAX_DEVICE_IDENTIFIER_STRING];
     char            Description[MAX_DEVICE_IDENTIFIER_STRING];
 
-#ifdef _WIN32
+#if defined(_WIN32) || defined(__EMSCRIPTEN__)
     LARGE_INTEGER   DriverVersion;            /* Defined for 32 bit components */
 #else
     DWORD           DriverVersionLowPart;     /* Defined for 16 bit driver components */
